File: Java - AWT - Swing/asteroids/Explosion.java
package asteroids;

import java.util.*;

class Explosion extends Entity {

    private Explosion() {
    }

    static public void spawn(Animation anim, float x, float y) {
        Queue<Explosion> pool = new LinkedList<>();

        Explosion asset = (pool.size() == 0 || pool.peek().active) ? new Explosion() : pool.remove();
        asset.anim = anim;
        asset.x = x;
        asset.y = y;
        asset.looped = false;
        asset.active = true;

        pool.add(asset);
        Astro.entities.add(asset);
    }

    public void update() {
        super.update();
        if (looped)
            active = false; // remove when the animation completes
    }
}

File: TypeScript - p5/src/Explosion.ts
import Astro from "./Astro";
import Animation from "./Animation";
import Entity from "./Entity";

export default class Explosion extends Entity {
  static spawn(anim: Animation, x: number, y: number) {
    const pool: Explosion[] = [];

    const asset = pool.length === 0 || pool[0].active ? new Explosion() : pool.shift()!;
    asset.anim = anim;
    asset.x = x;
    asset.y = y;
    asset.looped = false;
    asset.active = true;

    pool.push(asset);
    Astro.entities.push(asset);
  }

  update() {
    super.update();
    if (this.looped) this.active = false; // remove when the animation completes
  }
}
